extension/core_functions/scalar/math/numeric.cpp

```diff
@@ -566,6 +567,34 @@ struct RoundDecimalOperator {
 	}
 };
 
+struct RoundIntegerOperator {
+	template <class TA, class TB, class TR>
+	static inline TR Operation(TA input, TB precision) {
+		if (precision < 0) {
+			//	Do all the arithmetic at higher precision
+			using POWERS_OF_TEN_CLASS = typename DecimalCastTraits<TA>::POWERS_OF_TEN_CLASS;
+			if (-precision >= POWERS_OF_TEN_CLASS::CACHED_POWERS_OF_TEN) {
```

Comment:
What if precision cannot be negated (e.g. it is `INT_MIN` for type `int32`), can we use TryNegate here or otherwise check bounds prior to negating + add a test case for this scenario?

Reply:
Just swapping the signs on the compare does the trick!